packages/llvm.rb

```diff
@@ -2,38 +2,24 @@
 
 class Llvm < Package
   description 'The LLVM Project is a collection of modular and reusable compiler and toolchain technologies.'
-  homepage 'http://llvm.org/'
-  version '3.8.1-1'
-  source_url 'http://llvm.org/releases/3.8.1/llvm-3.8.1.src.tar.xz'
-  source_sha256 '6e82ce4adb54ff3afc18053d6981b6aed1406751b8742582ed50f04b5ab475f9'
+  homepage 'https://llvm.org/'
+  version '5.0.1'
+  source_url 'https://releases.llvm.org/5.0.1/llvm-5.0.1.src.tar.xz'
+  source_sha256 '5fa7489fc0225b11821cab0362f5813a05f2bcf2533e8a4ea9c9c860168807b0'
 
-  binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-3.8.1-1-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-3.8.1-1-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-3.8.1-1-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-3.8.1-1-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-    aarch64: '94255baa409229bbd0ad5f23981ee807fc39cb0c5160451daa0c3bfb1ebb78ae',
-     armv7l: '94255baa409229bbd0ad5f23981ee807fc39cb0c5160451daa0c3bfb1ebb78ae',
-       i686: '1d906912c090feeb81bb1777b8ff1ec7dcdadea08adea68c8a9cf9dfdcd8f2bb',
-     x86_64: 'efd72e40446eb7fb266b39a2e20df8398846ed76c96197b65dd31cb0fe023438',
-  })
-
-  depends_on 'buildessential'
-  depends_on 'cmake'
+  depends_on 'cmake' => :build
 
   def self.build
-    system "mkdir mybuilddir"
+    Dir.mkdir 'mybuilddir'
     Dir.chdir "mybuilddir" do
- 	    system "cmake .. -DLLVM_BUILD_LLVM_DYLIB=true"
-    	system "cmake --build ."
+ 	    system "cmake .. -DBUILD_SHARED_LIBS=true -DLLVM_TARGETS_TO_BUILD=\"X86\;ARM\" -DCMAKE_BUILD_TYPE=Release -DLLVM_OPTIMIZED_TABLEGEN=ON"
```

Comment:
I'm thinking we should leave out the `-DLLVM_TARGETS_TO_BUILD=\"X86\;ARM\"` option so this can be built on `x86_64` architecture also (the most popular Chromebook platform).

Reply:
Please change your indentation also.  If you are using tabs, please use spaces instead.  I know this is a bit nit picky but we strive for clean code.

Reply:
I'll change the indentation in a bit. The indentation is weird because I used just plain old vi to do it, because I broke my VIm/Spacemacs configuration by removing libclang over and over again. :/

The thing about the targets... I thought the same thing you did, but then I looked at CMakeLists.txt in Clang and noted this:

```

Reply:
Note the lack of an arch for x86_64 in the list. I haven't found any docs with state this explicitly, but I think the X86 target supports the x86_64 architecture.

I'll dig around more to confirm that. Anyway, the reason why I did it in the first place is that building all arches filled up my hard drive. I later discovered that the default mode for the build was set to debug, which made everything bigger, so it's possible we can turn all architectures back on without removing my computer's ability to build it.

It will take a lot longer to build though.

Reply:
Looks like aarch64 Chromebooks exist too. I'll just try to turn on all architectures and building again.